pytest_plugins/manual_skipped.py

```diff
@@ -9,33 +9,56 @@
 
 def pytest_configure(config):
     """Register custom marker for stubbed test cases."""
-    marker = "stubbed: Tests that are not automated yet or manual only."
-    config.addinivalue_line("markers", marker)
+    config.addinivalue_line('markers', 'stubbed: Tests that are not automated yet or manual only.')
 
 
 def pytest_collection_modifyitems(session, config, items):
-    """
-    Collects tests based on pytest option to select tests marked as failed on Report Portal
-    """
-    mark_passed = config.getvalue('mark_manuals_passed')
-    mark_skipped = config.getvalue('mark_manuals_skipped')
-    if mark_passed or (not mark_skipped):
-        return
+    """Remove/Include stubbed tests from collection based on CLI option"""
+    opt_passed = config.getvalue('mark_manuals_passed')
+    opt_skipped = config.getvalue('mark_manuals_skipped')
+    # TODO turn this into a flag or a choice option, this logic is just silly.
+    mark_skipped = opt_skipped and (not opt_passed)
+    include_stubbed = config.getvalue('include_stubbed')
+    selected = []
+    deselected = []
     for item in items:
-        if item.get_closest_marker(name='stubbed'):
-            item.add_marker(marker=pytest.mark.skip(reason=NOT_IMPLEMENTED))
+        stub_marked = item.get_closest_marker(name='stubbed')
+        # The test case is stubbed, and --include-stubbed was passed, include in collection
+        if stub_marked and include_stubbed:
+            selected.append(item)
+            # enforce skip/pass behavior by marking skip
+            if mark_skipped:
+                item.add_marker(marker=pytest.mark.skip(reason=NOT_IMPLEMENTED))
+            continue
+
+        # The test case is stubbed, but --include-stubbed was NOT passed, deselect the item
+        if stub_marked and not include_stubbed:
+            deselected.append(item)
+            continue
+
+        # Its a non-stubbed item, this hook doesn't apply
+        selected.append(item)
```

Comment:
```suggestion
        if stub_marked:
            if include_stubbed:
                selected.append(item)
                # enforce skip/pass behavior by marking skip
                if mark_skipped:
                    item.add_marker(marker=pytest.mark.skip(reason=NOT_IMPLEMENTED))
            else:
                deselected.append(item)
            continue
        
        # Its a non-stubbed item, this hook doesn't apply
        selected.append(item)
```

Reply:
@mshriver Modified the suggested code.

Reply:
The `continue` in your suggestion would prevent cases from getting collected if include_stubbed isn't true.

Reply:
@mshriver Thats what we want right, even your above code `not include_stubbed` is adding tests to deselected list, right ?

Reply:
The line 38 `selected.append(item)` is out of `if stub_marked` scope and continue hits only `if stub_marked` as its inside that loop !

Reply:
ah you're right. part of the reason I didn't nest it in the way you're suggesting is to reduce the amount of nested if statements.